website/docs/plugin/sdkv2/resources/write-only-attributes.mdx

```diff
@@ -0,0 +1,177 @@
+---
+page_title: Resources - Write-only Attributes
+description: Implementing write-only attributes within resources.
+---
+
+# Resources - Write-only Attributes
+
+~> **NOTE:** Write-only attributes are only supported in Terraform `v1.11` or higher
+
+Write-only attributes are managed resource attributes whose values are not saved to the Terraform plan or state artifacts. Write-only attributes
+should be used to handle secret values that do not need to be persisted in Terraform state, such as passwords, API keys, etc.
+The provider is expected to be the terminal point for an ephemeral value,
+which should either use the value by making the appropriate change to the API or ignore the value. Write-only attributes can accept [ephemeral values](/terraform/language/resources/ephemeral).
+
+## General Concepts
+
+The following are high level differences between regular attributes and write-only attributes:
+
+- Write-only attributes can accept ephemeral and non-ephemeral values
+
+- Write-only attribute values are only available in the configuration, and the prior state, planned state, and final state values for
+write-only attributes should always be `null`.
+    - Provider developers do not need to “clean up” write-only attribute values after using them as the SDKv2 will handle the nullification of write-only attributes for all RPCs.
+
+- Any value that is set for a write-only attribute using `(*ResourceData).Set()` by the provider will be set to `null` by SDKv2 before the RPC response is sent to Terraform
+
+- Write-only attribute values on its own will never cause a Terraform diff.
+    - This is because the prior state value for a write-only attribute will always be `null` and the planned/final state value will also be `null`, therefore, it cannot produce a diff on its own.
+    - The one exception to this case is if the write-only attribute is added to `requires_replace` via [CustomizeDiff](/terraform/plugin/sdkv2/resources/customizing-differences), in that case, the write-only attribute will always cause a diff/trigger a resource recreation
+
+- Since write-only attributes can accept ephemeral values, write-only attribute configuration values are not expected to be consistent between plan and apply.
+
+## Schema Behavior
+
+**Schema example:**
+
+```go
+"password_wo": {
+  Type:     schema.TypeString,
+  Required:  true,
+  WriteOnly: true,
+},
+```
+
+**Restrictions:**
+
+- Cannot be used in data source or provider schemas.
+- Must be set with either `Required` is `true` or `Optional` is `true`
+- Cannot be used when `Computed` is `true`
+- Cannot be used with aggregate schema types (e.g. `typeMap`, `typeList`, `typeSet`)
```

Comment:
Some considerations:
- Can't be used with `ForceNew`
- WriteOnly can't be set directly on an aggregate schema type, but it can be used within one (as long as it's not computed). We should make that distinction somewhere